src/main/java/org/opensearch/ad/task/ADBatchTaskCache.java

```diff
@@ -64,16 +73,23 @@
     private AtomicLong cacheMemorySize = new AtomicLong(0);
     private String cancelReason;
     private String cancelledBy;
+    private List<Entity> entity;
 
     protected ADBatchTaskCache(ADTask adTask) {
         this.detectorId = adTask.getDetectorId();
         this.taskId = adTask.getTaskId();
+        this.entity = adTask.getEntity() == null ? null : ImmutableList.copyOf(adTask.getEntity());
 
         AnomalyDetector detector = adTask.getDetector();
+        boolean isHC = detector.isMultientityDetector();
+        int numberOfTrees = isHC ? MULTI_ENTITY_NUM_TREES : NUM_TREES;
+        // use 1 for historical HC as realtime HC shingle size is hard coded as 1
```

Comment:
Do we have a centric place to enforce this hard code?

Reply:
No, realtime HC code https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/ml/EntityColdStarter.java#L273

@kaituo , you are removing this hard code to use shingle size configured in detector right?

Reply:
Sure, will change to `AnomalyDetectorSettings.DEFAULT_MULTI_ENTITY_SHINGLE`. 
Just confirm, from the comment https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java#L270, will change to 4? We will not support custom shingle size for HC detector?

Reply:
BTW, we should change `.dimensions(rcfNumFeatures)` to `.dimensions(AnomalyDetectorSettings.DEFAULT_MULTI_ENTITY_SHINGLE * rcfNumFeatures)` for this line 
 https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/ml/EntityColdStarter.java#L273, so if we will be safe if change the hard code shingle size to 4 in future.

Reply:
Talked with Kaituo, we may remove shingle(hide inside RCF model) in future, so will not have such hard code issue.